src/main/java/storage/Storage.java

```diff
@@ -0,0 +1,78 @@
+package storage;
+
+import exception.DukeException;
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.Todo;
+import task.TaskList;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+
+/**
+ * Interface that deals with pushing or pulling tasks from the storage.
+ */
+public class Storage {
+
+    String path;
+
+    public Storage(String path) {
+        this.path = path;
+    }
+
+    /**
+     * Retrieves tasks from the storage (if any) and adds them to a new task list.
+     * @return list of tasks retrieved from the storage, else an empty list.
+     * @throws DukeException If data file does not exist.
+     */
+    public ArrayList<Task> setUpData() throws DukeException {
+        try {
+            FileReader myObj = new FileReader(this.path);
+            BufferedReader br = new BufferedReader(myObj);
+            String line;
+            ArrayList<Task> tasks = new ArrayList<>();
+            while ((line = br.readLine()) != null) {
+                String[] lineArray = line.split("/");
+                Task task;
+                switch (lineArray[0]) {
```

Comment:
Indentation for switch is incorrect here, can update the code editor settings as per this module's coding standards

Reply:
Yup! Will do :) Thanks!